ssu.c

```diff
@@ -5,12 +5,55 @@
 #include <string.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <assert.h>
+#include <inttypes.h>
 #include <sys/wait.h>
 
 #ifndef ENV_PATH
 #define ENV_PATH "/bin:/sbin:/usr/bin:/usr/sbin"
 #endif
 
+#ifndef UID_MAX
+#define UID_MAX ((uint32_t)-1)
+#endif
+
+static const char *argv0;
+
+static uid_t parse_uid(const char *usr)
+{
+    intmax_t uid;
+    char *end;
+
+    uid = strtoimax(usr, &end, 10);
+
+    if (end == usr) {
+        fprintf(stderr, "%s: uid must be numeric: %s\n", argv0, usr);
+        exit(1);
+    }
+
+    if (end[0] != '\0') {
+        fprintf(stderr, "%s: invalid uid: %s\n", argv0, usr);
+        exit(1);
+    }
+
+    if (errno) {
```

Comment:
Maybe best to explicitly set `errno = 0` before calling `strtoimax`, in case something else mangled `errno`, and then check
```c
if (errno && uid == 0) {
```
to be sure that the error was from `strtoimax`.

Reply:
Done, thanks!